libqtile/command/base.py

```diff
@@ -40,6 +42,50 @@
     ItemT = Optional[tuple[bool, list[str | int]]]
 
 
+def expose_command(name: Callable | str | list[str] | None = None) -> Callable:
+    """
+    Decorator to expose commands to IPC.
+
+    The exposed command will have the name of the defined method.
+
+    Methods can also be exposed via multiple names by passing the names of
+    the additional method.
+
+    e.g. if a layout wants "up" and "previous" to call the
+    same method:
+
+    @expose_command("previous")
+    def up(self):
+        ...
+
+    `up` will be exposed as `up` and `previous`.
+
+    Multiple names can be passed as a list.
+    """
+
+    def wrapper(func: Callable):
+        setattr(func, "_cmd", True)
+        if name is not None:
+            if not hasattr(func, "_mapping"):
+                setattr(func, "_mapping", list())
+            if isinstance(name, list):
+                func._mapping += name  # type:ignore
+            elif isinstance(name, str):
+                func._mapping.append(name)  # type:ignore
+            else:
+                logger.error("Unexpected value received in command decorator: %s", name)
```

Comment:
I don't think this is necessary - the type signature states that only `str | list[str]` can be passed so this should never be reached
```suggestion
            else:
                func._mapping.append(name)  # type:ignore
```

Reply:
What if someone writes a custom widget and doesn't run `mypy` on their code?

It shouldn't happen with our codebase but other users may get this wrong.

Reply:
Yeah, fair enough